compiler/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -1210,6 +1194,58 @@ impl<'interner> TypeChecker<'interner> {
             }
         }
     }
+
+    /// Prerequisite: verify_trait_constraint of the operator's trait constraint.
+    ///
+    /// Although by this point the operator is expected to already have a trait impl,
+    /// we still need to match the operator's type against the method's instantiated type
+    /// to ensure the instantiation bindings are correct and the monomorphizer can
+    /// re-apply the needed bindings.
+    fn typecheck_operator_method(
+        &mut self,
+        expr_id: ExprId,
+        trait_method_id: TraitMethodId,
+        object_type: &Type,
+        span: Span,
+    ) {
+        let the_trait = self.interner.get_trait(trait_method_id.trait_id);
+
+        // The first (and only) method of each operator trait should be the operator method
```

Comment:
I don't think so - in rust some of the operator traits have additional optional methods to implement like `!=` if the user wants to implement that directly instead of relying on the default of `!(a == b)`. Perhaps I should update the comment though to remove the `(and only)` part to future proof it.

Reply:
Actually, I've removed the comment entirely since I swapped it to use the method index instead from Tom's suggestion.